File: Week_01/242.有效的字母异位词.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> dic = new HashMap<Character, Integer>();
        if (s.length() != t.length()) {
            return false;
        }
        for (char c: s.toCharArray()) {
            dic.put(c, dic.getOrDefault(c, 0) + 1);
        }
        for (char c: t.toCharArray()) {
            dic.put(c, dic.getOrDefault(c, 0) - 1);
            if (dic.get(c) < 0) {
                return false;
            }
        }
        return true;
    }
}

File: Week_01/242.有效的字母异位词.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        dic = {}
        if len(s) != len(t):
            return False
        for c in s:
            if c in dic.keys():
                dic[c] += 1
            else:
                dic[c] = 1
        for c in t:
            if c in dic.keys():
                dic[c] -= 1
                if dic[c] < 0:
                    return False
            else:
                return False
        return True